staging/src/k8s.io/cli-runtime/pkg/resource/builder.go

```diff
@@ -799,7 +833,11 @@ func (b *Builder) mappingFor(resourceOrKindArg string) (*meta.RESTMapping, error
 		return nil, err
 	}
 
-	return mapping, nil
+	if len(mappings) > 1 {
+		b.printWarning(resourceAmbiguousFmt, resourceOrKindArg)
```

Comment:
I haven't reviewed to be sure this printing technique is in keeping with printing warnings sent back from the kube-apiserver, but the idea of printing a warning the same way for ambiguous restmappings is good.

Reply:
This uses the new warning printer which was added in #108210, the goal of that PR is to start making alignments for the warnings printed by kubectl so I believe this is the best route to take, whether that aligns with API warnings or not I'm not 100% sure